code-studio/src/js/components/Grid.jsx

```diff
@@ -28,33 +45,40 @@ var Cell = React.createClass({
       classNames.push('highlighted');
     }
 
-    var tiles = ['border', 'path', 'start', 'end', 'obstacle'];
-    classNames.push(tiles[cell.tileType_]);
-
-    if (mazeUtils.isBeeSkin(this.props.skin)) {
-      var conditions = ['', 'flower-or-hive', 'flower-or-nothing', 'hive-or-nothing', 'flower-hive-or-nothing'];
-      var features = ['hive', 'flower'];
-      if (cell.isVariableCloud()) {
-        classNames.push('conditional');
-        classNames.push(conditions[cell.cloudType_]);
-      } else if (cell.featureType_ !== undefined) {
-        classNames.push(features[cell.featureType_]);
+    // Cell type-specific stuff
+    // TODO(elijah) This area contains a bunch of application-specific
+    // logic which would ideally live somewhere more flexible.
```

Comment:
I'm somewhat torn. On the one hand, this is levelbuilder-only code, so it doesn't need to be held to a super-high standard. On the other hand, this implementation is pretty gross.

I would lean toward leaving it as is with the TODO in there for now, for the dual reasons that improvements to levelbuilder-facing stuff is relatively low priority and that the only path to improvement that I see involves creating several subclasses of `Cell`, each with their own render logic, and that's its own kind of gross right there.